### PR TITLE
added ccsd initial state for free projection

### PR DESCRIPTION
**Pull request by ankit76**

### CODE REVIEW

File: ipie/addons/free_projection/propagation/CCSD.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# Authors: Fionn Malone <fmalone@google.com>
+#          Joonho Lee
+#
+
+from dataclasses import dataclass
+
+from ipie.utils.backend import arraylib as xp
+
+
+@dataclass
```
Comment (fdmalone) [inline ipie/addons/free_projection/propagation/CCSD.py]: It's a little weird to me that a dataclass  has an `__init__` method typically you would just have a factory method which does any instantiate it.

File: ipie/addons/free_projection/qmc/calc.py [hunk 2]
```diff
@@ -39,13 +40,15 @@ def build_fpafqmc_driver(
     estimator_filename: str = "estimates.0.h5",
     seed: int = None,
     qmc_options: dict = None,
+    ccsd: tuple = None,  # (t1, t2, orbital_rotation), orbital_rotation optional
```
Comment (fdmalone) [inline ipie/addons/free_projection/qmc/calc.py]: Should be Optional[tuple] = None.

File: ipie/addons/free_projection/qmc/calc.py [hunk 3]
```diff
@@ -145,6 +148,14 @@ def get_driver_fp(options: dict, comm: MPI.COMM_WORLD) -> FPAFQMC:
         ene_0 = local_energy(system, hamiltonian, walkers, trial)[0][0]
         propagator = FreePropagation(time_step=params.timestep, exp_nmax=10, ene_0=ene_0)
         propagator.build(hamiltonian, trial, walkers, mpi_handler)
+        ccsd = None
```
Comment (fdmalone) [inline ipie/addons/free_projection/qmc/calc.py]: Seems like this should be apart of walkers not it's own thing.

File: ipie/addons/free_projection/qmc/fp_afqmc.py [hunk 4]
```diff
@@ -49,10 +49,19 @@ def __init__(
...
+            system,
+            hamiltonian,
+            trial,
+            walkers,
+            propagator,
+            mpi_handler,
+            params,
+            verbose=verbose,
         )
+        self.ccsd = ccsd
```
Comment (fdmalone) [inline ipie/addons/free_projection/qmc/fp_afqmc.py]: why is this apart of AFQMC now? I think it's just initializing the walker wavefunction right? Just pass in the walkers or add a new factory which expects a CCSD object to initialize the walkers, or build it in the existing builder.

**Review by fdmalone — CHANGES_REQUESTED** [review]
Some small changes.

File: ipie/addons/free_projection/propagation/CCSD.py (hunk 1 above)
Comment (ankit76) [reply ipie/addons/free_projection/propagation/CCSD.py]: okay this is no longer a dataclass

File: ipie/addons/free_projection/qmc/fp_afqmc.py (hunk 4 above)
Comment (ankit76) [reply ipie/addons/free_projection/qmc/fp_afqmc.py]: My plan initially was to add an option for a general initial state, but then I added only ccsd. I agree it's a bit odd to have it here. I have moved it to the propagator which I think is more natural.

**Comment by linusjoonho** [comment]
@ankit76 Could you fix the failing tests so that we could merge this?

**Review by fdmalone — APPROVED** [review]

**Pull request merged**